tools/tesseratos/src/tesseratos/metrics_panel/plugin.cpp

```diff
@@ -0,0 +1,66 @@
+#include <imgui.h>
+
+#include <cubos/engine/imgui/plugin.hpp>
+#include <cubos/engine/imgui/serialization.hpp>
+
+#include <tesseratos/metrics_panel/plugin.hpp>
+
+using cubos::core::ecs::Read;
+using cubos::core::ecs::Write;
+using cubos::engine::Cubos;
+using cubos::engine::DeltaTime;
+
+using namespace tesseratos;
+
+struct Metrics
+{
+    float timeElapsed = 0.0;
+    float fps = 0.0;
+    int frameCounter = 0;
+    float secondTimer = 0.0;
+    // FPSGraph related
+    std::vector<float> fpsHistory;
+};
+
+static void panel(Read<DeltaTime> deltatime, Write<Metrics> metrics)
+{
+    ImGui::Begin("Metrics Panel");
+    // General Metrics
+    metrics->timeElapsed += deltatime->value;
+    metrics->frameCounter++;
+    metrics->secondTimer += deltatime->value;
+    if (metrics->secondTimer >= 1)
+    {
+        metrics->fps = static_cast<float>(metrics->frameCounter) / metrics->secondTimer;
+        metrics->secondTimer = 0.0;
+        metrics->frameCounter = 0;
+    }
+
+    // Graph
+    if (metrics->fpsHistory.size() > 100) // Seconds to Show
+    {
+        for (size_t i = 1; i < metrics->fpsHistory.size(); i++)
+        {
+            metrics->fpsHistory[i - 1] = metrics->fpsHistory[i];
+        }
+        metrics->fpsHistory[metrics->fpsHistory.size() - 1] = metrics->fps;
+    }
+    else
+    {
+        metrics->fpsHistory.push_back(metrics->fps);
+    }
+
```

Comment:
You are recording fps values per frame. It might be more useful if you record them per second. As it is, the graph zooms past information

Reply:
I think we should record all of them, but maybe instead of showing a graph we can just show:
- the minimum FPS in the last X seconds
- the maximum FPS in the last X seconds
- the average FPS in the last X seconds

Reply:
I think the graph is preferable, it gives more information. It just needs to be zoomed out a bit

Reply:
But this way we would miss resolution and possibly some spikes. I don't like that idea

Reply:
Hmm, we could have some more advanced math to register only the most significant fps measure within each second. Or we could just add more data point so we can zoom out without losing resolution.

Reply:
eu só meti gráfico pq eu usava o gráfico para ver no valorant ou no cs os fps